### PR TITLE
Change of default for allow_vendor_change

### PR DESCRIPTION
**Pull request by j-mracek**
Requires: https://github.com/rpm-software-management/dnf5/pull/621

### CODE REVIEW

File: doc/changes.rst [hunk 1]
```diff
@@ -143,6 +143,12 @@ Upgrade command
 Changes of configuration:
 =========================
 
+Defauilt of `allow_vendor_change` configuration option changed to `false`
```
Comment (m-blaha) [inline doc/changes.rst:146]: typo - defauilt -> default

File: doc/changes.rst [hunk 2]
```diff
@@ -143,6 +143,12 @@ Upgrade command
...
+-------------------------------------------------------------------------
+The new default value improves behavior for users that use third party repository (Copr). If they install a package from
+alternative provider, DNF will continue to consume updates from the same vendor. Additionally it enables to ship
+alternative streams using a different vendor.
+
```
Comment (m-blaha) [inline doc/changes.rst:151]:
```suggestion
The new default value improves the behavior for users of third-party repositories (e.g. Copr). If they install a package from
an alternative provider, DNF will continue to consume updates from the same vendor. Additionally it enables the shipping of
alternative streams using a different vendor.

```

**Comment by j-mracek** [comment]
It looks like tests failed on gpg import
```
       Importing PGP key 0x4429F6AC:
       Userid     : "dnf-ci-gpg-subkey"
       Fingerprint: A77F516D310F680AFBFFF814DEB7ECAB4429F6AC
       From       : file:///opt/ci/dnf-behave-tests/fixtures/gpgkeys/keys/dnf-ci-gpg-subkey/dnf-ci-gpg-subkey-public
      [org.rpm.dnf.v0.rpm.Rpm.TransactionError] rpm transaction failed with code 6.
```

**Comment by j-mracek** [comment]
I am unable to reproduce the test issue locally.

**Comment by m-blaha** [comment]
It looks like the gpg tests are a bit flaky... I'll re-run them, just to see whether the issue is reproducible.

**Comment by j-mracek** [comment]
The changes of unit tests are required because we have in tests newly build rpms with installed rpms injected by test case. In some cases new rpms have vendor set according to detected macros on your system.

**Pull request merged**